rtgui/perspective.cc

```diff
@@ -739,11 +741,29 @@ void PerspCorrection::setEditProvider(EditDataProvider* provider)
 
 void PerspCorrection::lineChanged(void)
 {
+    updateApplyDeleteButtons();
+
     if (listener) {
         listener->panelChanged(EvPerspControlLines, M("HISTORY_CHANGED"));
     }
 }
 
+void PerspCorrection::updateApplyDeleteButtons(void)
+{
+    if (batchMode) {
+        return;
+    }
+
+    bool edit_mode = lines_button_edit->get_active();
+    bool enough_lines = lines->getHorizontalCount() >= 2 || lines->getVerticalCount() >= 2;
+    const auto &tooltip = M("GENERAL_APPLY")
```

Comment:
Why `const &` on temporary? It's valid, but makes no sense here. If `set_tooltip_text()` takes the string by value, make it `auto` and `std::move()` it in the call, else make it `const auto`.